shell/platform/android/android_external_texture_gl.cc

```diff
@@ -62,19 +62,29 @@ void AndroidExternalTextureGL::Paint(SkCanvas& canvas,
     canvas.translate(bounds.x(), bounds.y());
     canvas.scale(bounds.width(), bounds.height());
     if (!transform.isIdentity()) {
-      SkMatrix transformAroundCenter(transform);
-
-      transformAroundCenter.preTranslate(-0.5, -0.5);
-      transformAroundCenter.postScale(1, -1);
-      transformAroundCenter.postTranslate(0.5, 0.5);
-      canvas.concat(transformAroundCenter);
+      sk_sp<SkShader> shader = image->makeShader(
+          SkTileMode::kRepeat, SkTileMode::kRepeat, sampling, transform);
+
+      SkPaint paint;
+      paint.setShader(shader);
+      canvas.drawRect(SkRect::MakeWH(1, 1), paint);
+    } else {
+      canvas.drawImage(image, 0, 0, sampling, nullptr);
     }
-    canvas.drawImage(image, 0, 0, sampling, nullptr);
   }
 }
 
 void AndroidExternalTextureGL::UpdateTransform() {
   jni_facade_->SurfaceTextureGetTransformMatrix(surface_texture_, transform);
+
+  SkMatrix inverted;
+  if (!transform.invert(&inverted)) {
+    FML_LOG(FATAL) << "Invalid SurfaceTexture transformation matrix";
+  }
+  transform = inverted;
+
+  transform.preScale(1.0, -1.0);
+  transform.postTranslate(0.0, -1.0);
```

Comment:
Would it be more appropriate to fix the math in the java layer?  It seems like keeping the lower level more dumb would be preferable, unless that is breaking user's code.